sqlmesh/core/model/meta.py

```diff
@@ -354,8 +354,9 @@ def unique_key(self) -> t.List[exp.Expression]:
 
     @property
     def partitioned_by(self) -> t.List[exp.Expression]:
+        """Columns to partition the model by, including the time column if it is not already included."""
         if self.time_column and self.time_column.column not in [
-            col for col in self._partition_by_columns
+            col for col in self.partitioned_by_user_columns
```

Comment:
Ah I see, this explains `partition_by_user_columns`.

Previously `_partition_by_columns` existed but was private, you made it public and exposed it as `partitioned_by_user_columns` to differentiate between columns the user set vs columns sqlmesh injected.

I wonder if this can be cleaned up in future by having engine adapters indicating whether or not its appropriate to include the time_column in the list of columns to partition by. Then we might be able to use a single property again

Reply:
Ah, that's a cleaner solution for sure - should be easy to add

Reply:
Hmm...meta.py doesn't have access to the engine adapter here

Reply:
Another approach would be to convert partitioned_by into a dict, with user columns and time columns stored separately.

But that would mean collapsing it into a list everywhere it's used.

Reply:
Yeah I think your solution is correct given the current design :+1: